AppServer/google/appengine/api/logservice/logservice_stub.py

```diff
@@ -14,339 +14,159 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+"""Stub implementation for Log Service that uses sqlite."""
 
-
-
-
-"""Stub implementation for Log Service that utilizes the Datastore.
-
-Logs can be flushed, which will store them in the Datastore, and retrieved for
-use by the user. Users can retrieve logs along a number of different query
-parameters, including the time the request began, whether or not
-application-level logs should be included, and so on.
-"""
-
-
-
-
-import httplib
-import os
 import time
-import threading
+import os
+import socket
+import struct
+import logging
+import base64
+from Queue import Queue, Empty
+from collections import defaultdict
+
+import capnp # pylint: disable=unused-import
 
 from google.appengine.api import apiproxy_stub
-from google.appengine.api import datastore_errors
-from google.appengine.api import logservice
-from google.appengine.api import namespace_manager
 from google.appengine.api.logservice import log_service_pb
-from google.appengine.api.logservice import logservice
-from google.appengine.ext import db
 from google.appengine.runtime import apiproxy_errors
 
-try:
-  import json
-except ImportError:
-  import simplejson as json
-
-LOG_NAMESPACE = '_Logs'
-_FUTURE_TIME = 2**34
-_REQUEST_TIME = 0
-_CURRENT_REQUEST_ID_HASH = ''
-
-
-def _get_request_id():
-  """Returns the request ID bound to this request.
+import logging_capnp
 
-  Specifically, we see if the request ID hash has changed since the last time we
-  have examined it. If so, we generate a new ID based on the current time.
-  Regardless, we return a string whose value decreases w.r.t. time, so that
-  values stored in the Datastore will be sorted from newest to oldest.
-  """
-  global _CURRENT_REQUEST_ID_HASH
-  global _REQUEST_TIME
-  request_id_hash = os.environ.get('REQUEST_ID_HASH', '')
+_I_SIZE = struct.calcsize('I')
 
-  if _CURRENT_REQUEST_ID_HASH != request_id_hash:
-    _CURRENT_REQUEST_ID_HASH = request_id_hash
-    _REQUEST_TIME = time.time()
-
-  return str(int((_FUTURE_TIME - _REQUEST_TIME) * 1000000))
 
+def _cleanup_logserver_connection(connection):
+  try:
+    connection.close()
+  except socker.error:
```

Comment:
This is a typo correct? I've caught it several times and changed it but it appeared to get lost on my PR.